scipy/meson.build

```diff
@@ -321,6 +311,17 @@ fortran_ignore_warnings = ff.get_supported_arguments(
  _fflag_Wno_tabs,
 )
 
+# Intel Fortran (ifort) does not run the preprocessor by default, if Fortran
+# code uses preprocessor statements, add this compile flag to it.
+_fflag_fpp = []
+if ff.get_id() == 'intel-cl'
+  if is_windows
+    _fflag_fpp = ff.get_supported_arguments('/fpp')
+  else
+    _fflag_fpp = ff.get_supported_arguments('-fpp')
```

Comment:
I'll just note that on a Linux node with Intel (classic) compiler toolchain loaded (`ifort (IFORT) 2021.9.0 20230302`) I have a successful `dev.py` build on `main` as well as on this branch. I was hoping to reproduce the original build error on `main`, but perhaps there is some other sensitivity that I don't get on a supercomputer node, dunno... and also I saw Windows mentioned in the other issue.

Since CI is green and the changes seem useful, +1 from me at least.

Oh, and there are test failures (even segfaults) when I test with the Intel toolchain, but those are present on `main` and this branch, and could also relate to specific OpenBLAS buildout I have I suppose. Anyway, that seems out of scope here.

Reply:
Thanks for checking. I suspect it's Windows-specific indeed. Re segfaults: if you use Intel Fortran, you should build against MKL I think, that's the relevant thing and if there are still segfaults then, I'll get worried. For Intel Fortran + OpenBLAS I have no idea what to expect.